File: rb101/rb101-rb109-small_problems/easy9/swap_name.rb
def swap_name(name)
  name.split.reverse.join(', ')
end

p swap_name('Joe Roberts') == 'Roberts, Joe'

# ALTERNATIVE SOLUTION:

# The below solution specifies a pattern argument for #split.
# According to its documentation, #split will automatically
# use the whitespace argument ' ' when no pattern (or nil) is
# specified:

# def swap_name(name)
#   name.split(' ').reverse.join(', ')
# end

File: js210/js210-small_problems/easy5/swapName.js
"use strict";

function swapName(name) {
  return name.split(' ').reverse().join(', ');
}

console.log(swapName('Joe Roberts'));    // "Roberts, Joe"


// FURTHER EXPLORATION:

// function swapName(name) {
//   const names = name.split(' ');
//   const lastName = names.pop();
//   const firstNames = names.join(' ');

//   return `${lastName}, ${firstNames}`;
// }

// console.log(swapName('Ricky Bobby Roberts'));
